api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -184,13 +184,13 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			}
 			
 			//concept should be the same as on previous order, same applies to drug for drug orders
-			boolean isDrugOrderAndHasADrug = isDrugOrder && ((DrugOrder) order).getDrug() != null;
+			boolean isDrugOrderAndHasADrug = isDrugOrder && (((DrugOrder) order).getDrug() != null || ((DrugOrder) order).isNonCodedDrug()) ;
 			if (!OpenmrsUtil.nullSafeEquals(order.getConcept(), previousOrder.getConcept())) {
 				throw new APIException("The concept of the previous order and the new one order don't match");
 			} else if (isDrugOrderAndHasADrug) {
 				DrugOrder drugOrder1 = (DrugOrder) order;
 				DrugOrder drugOrder2 = (DrugOrder) previousOrder;
-				if (!OpenmrsUtil.nullSafeEquals(drugOrder1.getDrug(), drugOrder2.getDrug())) {
+				if (!drugOrder1.hasSameOrderableAs(drugOrder2)) {
```

Comment:
since order need not be drug order always line 188 should still remain the same..
(or) we should call order.hasSameOrderableAs() and drugOrder.hasSameOrderableAs()